src/node_internals.h

```diff
@@ -75,6 +75,100 @@ struct sockaddr;
+    V(serdes,         MODREG_WITH_DEF)                                        \
+    V(url,            MODREG_WITH_DEF)                                        \
+    V(util,           MODREG_WITH_DEF)                                        \
+    V(v8,             MODREG_WITH_DEF)                                        \
+    V(zlib,           MODREG_WITH_DEF)                                        \
+    V(icu,            ICU_MODREG_DEF)                                         \
+    V(pipe_wrap,      MODREG_WITH_DEF)                                        \
+    V(process_wrap,   MODREG_WITH_DEF)                                        \
+    V(signal_wrap,    MODREG_WITH_DEF)                                        \
+    V(spawn_sync,     MODREG_WITH_DEF)                                        \
+    V(stream_wrap,    MODREG_WITH_DEF)                                        \
+    V(tcp_wrap,       MODREG_WITH_DEF)                                        \
+    V(timer_wrap,     MODREG_WITH_DEF)                                        \
+    V(tty_wrap,       MODREG_WITH_DEF)                                        \
+    V(udp_wrap,       MODREG_WITH_DEF)                                        \
+    V(uv,             MODREG_WITH_DEF)                                        \
+    V(inspector,      MODREG_WITH_DEF)                                        \
+    V(crypto,         OPENSSL_MODREG_DEF)                                     \
+    V(tls_wrap,       OPENSSL_MODREG_DEF)
+
+/* for nodejs built-in modules only */
+#define NODE_MODULE_CONTEXT_AWARE_X_BUILTIN(modname,                          \
+                                            regfunc,                          \
+                                            priv,                             \
+                                            flags)                            \
+  static node::node_module _module =                                          \
+  {                                                                           \
+    NODE_MODULE_VERSION,                                                      \
+    flags,                                                                    \
+    NULL,                                                                     \
+    __FILE__,                                                                 \
+    NULL,                                                                     \
+    (node::addon_context_register_func) (regfunc),                            \
+    NODE_STRINGIFY(modname),                                                  \
+    priv,                                                                     \
+    NULL                                                                      \
+  };                                                                          \
+  NODE_C_CTOR_BUILTIN(_register_ ## modname) {                                \
+    node_module_register(&_module);                                           \
```

Comment:
Another something for a future cleanup PR: export the struct and call `node_module_register()` directly in node.cc, instead of first going through `_register_foo()`.

Performance-wise it's unlikely to matter but it should make it a little easier to understand for future maintainers.

Also, I think you can get rid of `NODE_C_CTOR_BUILTIN` now, you don't need to special-case for Windows anymore.